sycl/source/backend.cpp

```diff
@@ -130,8 +130,12 @@ __SYCL_EXPORT event make_event(pi_native_handle NativeHandle,
   Plugin.call<PiApiKind::piextEventCreateWithNativeHandle>(
       NativeHandle, ContextImpl->getHandleRef(), !KeepOwnership, &PiEvent);
 
-  return detail::createSyclObjFromImpl<event>(
+  event Event = detail::createSyclObjFromImpl<event>(
       std::make_shared<event_impl>(PiEvent, Context));
+
+  // We've transferred ownership from PiEvent into Event.
```

Comment:
The L0 handle "ownership" belongs to either 1) the app or 2) the SYCL RT and it is controlled by `KeepOwnership` argument. So, this comment is not clear as is. Please also explain why do we need to release a reference right after we've created it.

Reply:
In the comments throughout this patch "ownership" is used as in `std::shared_ptr`'s ownership, which means that there might be multiple pointers/handles sharing the ownership of an object. In particular, the code here could semantically be this:

```
  auto PiEvent = make_shared_resource<pi::PiEvent>(Plugin.call<>(...));
  event Event = detail::createSyclObjFromImpl<event>(...std::move(PiEvent), ...);
```

We don't have such `shared_resource` that we can `std::move`, so we have to decrement reference counter manually.